tiledb/sm/query/dense_tiler.cc

```diff
@@ -200,166 +200,107 @@ Status DenseTiler<T>::get_tile(
+    std::vector<uint8_t> fill_var(sizeof(uint64_t), 0);
+
+    // Initialize position tile
+    Tile tile_pos;
+    RETURN_NOT_OK(tile_pos.init_unfiltered(
+        constants::format_version,
+        constants::cell_var_offset_type,
+        tile_off_size,
+        constants::cell_var_offset_size,
+        0));
+
+    // Fill entire tile with MAX_UINT64
+    std::vector<uint64_t> to_write(
+        cell_num_in_tile, std::numeric_limits<uint64_t>::max());
+    RETURN_NOT_OK(tile_pos.write(to_write.data(), 0, tile_off_size));
+    to_write.clear();
+    to_write.shrink_to_fit();
+
+    // Get position tile
+    auto cell_num_in_buff =  // TODO: fix
+        (buff_off_size - (offsets_extra_element_ * offsets_bytesize_)) /
+        offsets_bytesize_;
+    std::vector<uint64_t> cell_pos(cell_num_in_buff);
+    for (uint64_t i = 0; i < cell_num_in_buff; ++i)
+      cell_pos[i] = i;
+    RETURN_NOT_OK(copy_tile(
+        id, constants::cell_var_offset_size, (uint8_t*)&cell_pos[0], tile_pos));
+
+    // Copy real offsets and values to the corresponding tiles
+    void* tile_pos_buff_tmp = tile_pos.data();
+    auto tile_pos_buff = (uint64_t*)tile_pos_buff_tmp;
+    uint64_t tile_off_offset = 0, offset = 0, val_offset, val_size, pos;
+    auto mul = (offsets_format_mode_ == "bytes") ? 1 : cell_size;
+    auto& tile_off = tile.offset_tile();
+    auto& tile_val = tile.var_tile();
+    for (uint64_t i = 0; i < cell_num_in_tile; ++i) {
+      pos = tile_pos_buff[i];
+      RETURN_NOT_OK(tile_off.write(&offset, tile_off_offset, sizeof(offset)));
+      tile_off_offset += sizeof(offset);
+      if (pos == std::numeric_limits<uint64_t>::max()) {  // Empty
+        RETURN_NOT_OK(tile_val.write_var(&fill_var[0], offset, cell_size));
+        offset += cell_size;
+      } else {  // Non-empty
+        val_offset = ((offsets_bytesize_ == 8) ?
+                          ((uint64_t*)buff_it->second.buffer_)[pos] :
+                          ((uint32_t*)buff_it->second.buffer_)[pos]) *
+                     mul;
+        val_size = (pos < cell_num_in_buff - 1) ?
+                       (((offsets_bytesize_ == 8) ?
+                             ((uint64_t*)buff_it->second.buffer_)[pos + 1] :
+                             ((uint32_t*)buff_it->second.buffer_)[pos + 1]) *
+                            mul -
+                        val_offset) :
+                       buff_var_size - val_offset;
+        RETURN_NOT_OK(
+            tile_val.write_var(&buff_var[val_offset], offset, val_size));
+        offset += val_size;
+      }
+    }
 
-  // Copy tile from buffer
-  RETURN_NOT_OK(copy_tile(id, cell_size, buff, tile));
+    tile_val.set_size(offset);
+  } else {
+    auto cell_size = array_schema_.cell_size(name);
+    auto tile_size = cell_size * cell_num_in_tile;
+    auto buff = (uint8_t*)buffers_->find(name)->second.buffer_;
+    assert(buff != nullptr);
 
-  return Status::Ok();
-}
+    memset(tile.fixed_tile().data(), 0, tile_size);
```

Comment:
this means that `init_unfiltered` can safely be replaced by `memset` in those cases? We don't need to set anymore all the rest of fields that we set in `init_unfiltered`?

Reply:
Not exactly... If you look at tile, there's a fill_with_zeros member on the constructor... And the ordered writer would initialize some tiles with that flag for those tiles, I added the memset. Init_unfiltered is still called, but it's called when the WriterTile objects get created inside of the batches.

Reply:
We can probably now remove fill_with_zeros... The reason I don't want to use it is because for offsets tiles for examples, the ordered writer initializes the offsets with 0xFF values... So I'd rather have all initialization code, in the same place for now, and possibly move it later.